start.sh

```diff
@@ -3,6 +3,8 @@
 # Chown volumes so that nginx can read/write
 chown www-data:www-data /usr/share/nginx/html/images
 chown www-data:www-data /usr/share/nginx/html/LocalSettings.php
+chown -R www-data:www-data /tmp/extensions/
+cp -r /tmp/extensions/* /usr/share/nginx/html/extensions/
```

Comment:
Yeah, I agree however if you are doing a `cp`, you are better off having a seperate Dockerfile that simply does `ADD` to the extension directory and using this image as a base.

I think the best solution for this is to either mount via the subdirectory in `extensions` or to create a new Dockerfile that adds your extensions into there. This extension volume only leads to a violation of the [immutability](https://docs.docker.com/articles/dockerfile_best-practices/) (see ephemeral section) of this image, which is what Docker leads itself to be.

My suggestion is to do something like this:

       FROM simplyintricate/mediawiki

        ENV BASEDIR /usr/share/nginx/html

        # Extensions
        WORKDIR ${BASEDIR}/extensions
        RUN apt-get update && apt-get install -y git

        RUN git clone https://gerrit.wikimedia.org/r/p/mediawiki/extensions/Mantle.git
        RUN cd Mantle && git checkout REL1_24
        RUN git clone https://gerrit.wikimedia.org/r/p/mediawiki/extensions/MobileFrontend.git
        RUN cd MobileFrontend && git checkout REL1_24

This would really allow you to start a container without needing to even mount a volume, which will help preserve the immutability of this container. It's also just about the same approach as what you're doing with the `cp` command.  This also preserves the location of the extensions since you don't run into the volume issue you're talking about earlier. 

If you still want to mount the extensions by volume, you could also consider moving the MediaWiki provided extensions to a different directory and then when the container starts, move them back into the extensions directory. However, I don't like the approach of mounting a volume either way due to the immutability problem.